webviz_subsurface/containers/_intersect.py

```diff
@@ -0,0 +1,297 @@
+from glob import glob
+from pathlib import PurePath as path
+from collections import OrderedDict
+import numpy as np
+import pandas as pd
+import dash_html_components as html
+import dash_core_components as dcc
+from dash.dependencies import Input, Output, State
+from dash_table import DataTable
+from webviz_config.common_cache import cache
+from ..datainput import scratch_ensemble, load_surface, get_wfence, get_hfence
+
+
+class Intersect():
+    '''Simple example to calculate a well fence using Xtgeo
+    and plotting using Plotly graph object'''
+
+    def __init__(self, app, container_settings, ensemble,
+                 well_path, surface_cat, surface_names,
+                 well_suffix='.rmswell'):
+
+        self.well_path = well_path
+        self.ensemble_path = container_settings['scratch_ensembles'][ensemble]
+        self.ensemble = scratch_ensemble(ensemble, self.ensemble_path)
+        self.well_suffix = well_suffix
+        self.surface_cat = surface_cat
+        self.surface_names = surface_names
+        self.set_callbacks(app)
+
+    @property
+    def realizations(self):
+        d = {self.ensemble._realizations[real]._origpath: real
+             for real in self.ensemble._realizations}
+        return OrderedDict(sorted(d.items()))
+
+    @property
+    def well_names(self):
+        well_folder = f'{self.well_path}/*{self.well_suffix}'
+        return sorted([well for well in glob(well_folder)])
+
+    @property
+    def colors(self):
+        return ['#543005', '#8c510a', '#bf812d', '#dfc27d',
+                '#f6e8c3', '#f5f5f5', '#c7eae5', '#80cdc1',
+                '#35978f', '#01665e', '#003c30', '#543005',
+                '#8c510a', '#bf812d', '#dfc27d', '#f6e8c3',
+                '#f5f5f5', '#c7eae5', '#80cdc1',
+                '#35978f', '#01665e', '#003c30',  '#543005',
+                '#8c510a', '#bf812d', '#dfc27d', '#f6e8c3',
+                '#f5f5f5', '#c7eae5', '#80cdc1',
+                '#35978f', '#01665e', '#003c30']
+
+    @property
+    def surface_colors(self):
+        return {surf: self.colors[i]
+                for i, surf in enumerate(self.surface_names)}
```

Comment:
Will error in (the unlikely) event of more surface names than defined colors? Perhaps implement a color cycling like matplotlib? E.g. `self.colors[i % len(self.colors)]` instead of `self.colors[i]`.